src/cnfizers/Tseitin.cc

```diff
@@ -1,357 +1,160 @@
-LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
-WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-*********************************************************************/
-
-#include "Cnfizer.h"
 #include "Tseitin.h"
 
-
-//
-// Performs the actual cnfization
-//
-
-
-bool Tseitin::cnfize(PTRef term) {
-    bool res = true;
+void Tseitin::cnfize(PTRef term) {
     vec<PTRef> unprocessed_terms {term};
 
-    //
     // Visit the DAG of the formula
-    //
     while (unprocessed_terms.size() != 0) {
-
         PTRef ptr = unprocessed_terms.last();
         unprocessed_terms.pop();
-        //
         // Skip if the node has already been processed before
-        //
-        if (alreadyCnfized.contains(ptr, current_frame_term)){
+        if (alreadyCnfized.contains(ptr, currentFrameId)){
             continue;
         }
 
         // Here (after the checks) not safe to use Pterm& since cnfize.* can alter the table of terms
         // by calling findLit
         int sz = logic.getPterm(ptr).size();
         if (logic.isAnd(ptr))
-            res &= cnfizeAnd(ptr);
+            cnfizeAnd(ptr);
         else if (logic.isOr(ptr))
-            res &= cnfizeOr(ptr);
+            cnfizeOr(ptr);
         else if (logic.isXor(ptr))
-            res &= cnfizeXor(ptr);
+            cnfizeXor(ptr);
         else if (logic.isIff(ptr))
-            res &= cnfizeIff(ptr);
+            cnfizeIff(ptr);
         else if (logic.isImplies(ptr))
-            res &= cnfizeImplies(ptr);
+            cnfizeImplies(ptr);
         // Ites are handled through the ite manager system and treated here as variables
 //        else if (logic.isIte(ptr))
 //            res &= cnfizeIfthenelse(ptr);
-        else if (!logic.isNot(ptr) && sz > 0) {
+        else if (!logic.isNot(ptr) && sz > 0) { // do not recurse into atoms
             goto tseitin_end;
         }
-        {
-            Pterm const& pt = logic.getPterm(ptr);
-            for (int i = 0; i < pt.size(); i++) {
-                unprocessed_terms.push(pt[i]); // Using the PTRef is safe if a reallocation happened
-            }
+        for (PTRef child : logic.getPterm(ptr)) {
+            unprocessed_terms.push(child);
         }
 tseitin_end:
-        alreadyCnfized.insert(ptr, current_frame_term);
+        alreadyCnfized.insert(ptr, currentFrameId);
     }
-
-    return res;
 }
 
-
-bool Tseitin::cnfizeAnd(PTRef and_term)
-{
-//  assert( list );
-//  assert( list->isList( ) );
-    //
+void Tseitin::cnfizeAnd(PTRef and_term) {
     // ( a_0 & ... & a_{n-1} )
-    //
     // <=>
-    //
     // aux = ( -aux | a_0 ) & ... & ( -aux | a_{n-1} ) & ( aux & -a_0 & ... & -a_{n-1} )
-    //
     Lit v = this->getOrCreateLiteralFor(and_term);
-    vec<Lit> little_clause;
     vec<Lit> big_clause;
     int size = logic.getPterm(and_term).size();
     big_clause.capacity(size + 1);
-    little_clause.push(~v);
```

Comment:
Why remove little clause(or rather why it was there, because it looks like a data duplication)